lib/friendly_id/sequential_slug_generator.rb

```diff
@@ -0,0 +1,53 @@
+module FriendlyId
+  class SequentialSlugGenerator
+    attr_accessor :scope, :slug, :slug_column, :sequence_separator
+
+    def initialize(scope, slug, slug_column, sequence_separator)
+      @scope = scope
+      @slug = slug
+      @slug_column = slug_column
+      @sequence_separator = sequence_separator
+    end
+
+    def generate
+      slug + sequence_separator + next_sequence_number.to_s
+    end
+
+  private
+
+    def next_sequence_number
+      if last_sequence_number == 0
+        2
+      else
+        last_sequence_number + 1
+      end
+    end
+
+    def last_sequence_number
+      slug_conflicts.last.split("#{slug}#{sequence_separator}").last.to_i
```

Comment:
Some of my models generate slugs from strings containing integers only, like "1234". This code fails on such case, therefore I'd propose to replace to_i with something else, or change the logic to make it similar to friendly-id 4 code

    def last_sequence_number
      begin
        Integer(slug_conflicts.last.split("#{slug}#{sequence_separator}").last)
      rescue
        0
      end
    end